src/app-layout/runtime-api.tsx

```diff
@@ -2,40 +2,64 @@
 // SPDX-License-Identifier: Apache-2.0
 import React from 'react';
 
-import { fireNonCancelableEvent } from '../internal/events';
+import { fireNonCancelableEvent, NonCancelableEventHandler } from '../internal/events';
 import { DrawerConfig as RuntimeDrawerConfig } from '../internal/plugins/controllers/drawers';
 import { RuntimeContentWrapper } from '../internal/plugins/helpers';
 import { sortByPriority } from '../internal/plugins/helpers/utils';
+import VisibilityStateManager from '../internal/plugins/helpers/visibility-state-manager';
 import { AppLayoutProps } from './interfaces';
 
 export interface DrawersLayout {
   before: Array<AppLayoutProps.Drawer>;
   after: Array<AppLayoutProps.Drawer>;
 }
 
+const visibilityStateManagerMap = new Map<string, VisibilityStateManager>();
+
 export function convertRuntimeDrawers(drawers: Array<RuntimeDrawerConfig>): DrawersLayout {
   const converted = drawers.map(
     ({
       mountContent,
       unmountContent,
       trigger,
       ...runtimeDrawer
-    }): AppLayoutProps.Drawer & { orderPriority?: number } => ({
-      ...runtimeDrawer,
-      ariaLabels: { drawerName: runtimeDrawer.ariaLabels.content ?? '', ...runtimeDrawer.ariaLabels },
-      trigger: {
-        iconSvg: (
-          // eslint-disable-next-line react/no-danger
-          <span dangerouslySetInnerHTML={{ __html: trigger.iconSvg }} />
+    }): AppLayoutProps.Drawer & {
+      orderPriority?: number;
+      onShow?: NonCancelableEventHandler;
+      onHide?: NonCancelableEventHandler;
+    } => {
+      let visibilityStateManager: VisibilityStateManager;
+      if (visibilityStateManagerMap.has(runtimeDrawer.id)) {
```

Comment:
The purpose of this code is to keep the same class instance after components' re-rendering